### PR TITLE
Combine coverage data with GitHub Action

### LINKED ISSUE #468: Coverage check on GA does not combine coverage data
On GitHub Action, each test run triggers the coverage session. Due to the fact that the workflow runs every session and interpreter separately, coverage can fail in CI even though the combined coverage would have reached the target, and local tests succeeded. By contrast, the Codecov check will succeed and combine coverage across interpreters (and platforms).

Example: https://github.com/cjolowicz/cookiecutter-hypermodern-python-instance/pull/324

One way to fix this is to never fail the workflow from coverage, and rely on Codecov exclusively. Ideally, we would still produce a terminal report with Coverage.py, just without failing the job. An open question is how well we can ensure that local testing and CI agree in their final verdict on code coverage. (For example, note that local testing does not combine coverage across platforms, while Codecov does.)

### PR DESCRIPTION
**Pull request by oncleben31**
- In `tests` nox session, `coverage` session is automatically called only in interactive mode.
- In `tests` GA workflow, coverage data are saved as artefact for each entry in the matrix.
- In `tests` GA workflow, a `coverage`job is added to download from artefacts all coverage data, combine them and upload the report on codecov.
- Documentation update.

Closes #468

### CODE REVIEW

**Comment by oncleben31** [comment]
@cjolowicz do you have advise for the best way to test my modifications before merging?

**Comment by oncleben31** [comment]
I have test in the repo: https://github.com/oncleben31/cookiecutter-hypermodern-python-instance
coverage part is working well.

It was a good exercise to use cookiecutter, pipx, and cupper.

**Review by cjolowicz — CHANGES_REQUESTED** [review]
Many thanks for your work, perfect! I left some inline comments with suggestions. (These are very minor points. I can also address them in the merge, if you don't get around to it yourself.) 

And sorry for not being able to help out with testing earlier. I see you figured out a solution. That's exactly how I do it as well.

File: {{cookiecutter.project_name}}/.github/workflows/tests.yml [hunk 1]
```diff
@@ -80,18 +80,59 @@ jobs:
...
+        run: |
+          pip install --constraint=.github/workflows/constraints.txt poetry
+          poetry --version
+
+      - name: Install Nox
+        run: |
+          pip install --constraint=.github/workflows/constraints.txt nox
+          nox --version
+
+      - name: "Download coverage data"
```
[the remark this answers is not in the document]
Comment (oncleben31) [reply {{cookiecutter.project_name}}/.github/workflows/tests.yml]: Done.

File: docs/guide.rst [hunk 2]
```diff
@@ -2303,8 +2305,9 @@ The Tests workflow uses the following GitHub Actions:
```
[the remark this answers is not in the document]
Comment (oncleben31) [reply docs/guide.rst:2305]: Done

File: {{cookiecutter.project_name}}/.github/workflows/tests.yml [hunk 3]
```diff
@@ -80,18 +80,59 @@ jobs:
         run: |
           nox --force-color --python=${{ "{{" }} matrix.python-version {{ "}}" }}
 
+      - name: Upload coverage data
+        if: always() && matrix.session == 'tests'
+        uses: "actions/upload-artifact@v2"
+        with:
+          name: coverage-data
+          path: {{ "${{ github.workspace }}/.coverage.*" }}
```
[the remark this answers is not in the document]
Comment (oncleben31) [reply {{cookiecutter.project_name}}/.github/workflows/tests.yml]: You are right all files have a different name.

**Comment by oncleben31** [comment]
Thank you for your detailed feedbacks. They are all taken into account.

**Review by cjolowicz — APPROVED** [review]
Thanks!

Here's a passing and a failing test (with and without your changes, respectively):

- https://github.com/cjolowicz/cookiecutter-hypermodern-python-instance/pull/323
- https://github.com/cjolowicz/cookiecutter-hypermodern-python-instance/pull/324

**Pull request merged**